src/a01/commands/get_run.py

```diff
@@ -25,37 +22,36 @@
           'source code.')
 @arg('raw', help='For debug.')
 async def get_run(run_id: str, log: bool = False, recording: bool = False, recording_az_mode: bool = False,
-                  show_all: bool = False, raw: bool = False) -> None:
+                  show_all: bool = False, raw: bool = False) -> CommandOutput:
     logger = logging.getLogger(__name__)
+    output = SequentialOutput()
 
     try:
         async with AsyncSession() as session:
             tasks = [Task.from_dict(each) for each in await session.get_json(f'run/{run_id}/tasks')]
-            tasks = TaskCollection(tasks, run_id)
+            tasks_output = TasksOutput(tasks, show_all)
 
-            output_in_table(tasks.get_table_view(failed=not show_all), headers=tasks.get_table_header())
-            output_in_table(tasks.get_summary(), tablefmt='plain')
+            output.append(tasks_output)
+            output.append(TasksSummary(tasks))
 
             if log:
-                for failure in tasks.get_failed_tasks():
-                    output_in_table(zip_longest(failure.get_table_header(), failure.get_table_view()), tablefmt='plain')
-                    output_in_table(await get_log_content_async(failure.log_resource_uri, session),
-                                    tablefmt='plain',
-                                    foreground_color=colorama.Fore.CYAN)
-                output_in_table(tasks.get_summary(), tablefmt='plain')
+                for failure in tasks_output.get_failed_tasks():
+                    output.append(TaskBriefOutput(failure))
+                    output.append(TaskLogOutput(await get_log_content_async(failure.log_resource_uri, session)))
+                output.append(TasksSummary(tasks))
```

Comment:
>output.append(TasksSummary(tasks)) [](start = 16, length = 34)

Do you still need this if it already exists in line 35?

Reply:
Yes. It is intended to have the output duplicate. The reason is that the output could be long so this save users from scroll back to the top.